packages/react-forms/src/InputDate/index.js

```diff
@@ -21,141 +21,173 @@ function toYYYYMMDD(date) {
+  isOpen: isOpenProp,
+  onToggle,
+  min,
+  max,
+  onCalendarChange,
+  calendarValue,
+  ...props
+}: Props) => {
+  const component = createRef();
+  const refA = createRef();
+  const refB = createRef();
+
+  const [current, setCurrent] = useState(calendarValue);
+  const [isOpen, setOpen] = useState(
+    isBoolean(isOpenProp) ? isOpenProp : false
+  );
+
+  const setIsOpen = useCallback(
+    (isOpen: boolean) => {
+      if (onToggle) {
+        onToggle(isOpen);
+      } else if (!isBoolean(isOpenProp)) {
+        setOpen(isOpen);
+      }
+    },
+    [onToggle, setOpen, isOpenProp]
+  );
+
+  const handleOpen = useCallback(() => {
+    // istanbul ignore else
+    if (isOpen === false) {
+      setIsOpen(true);
     }
-  };
+  }, [isOpen, setIsOpen]);
 
-  handleInputChange = (evt: SyntheticInputEvent<HTMLInputElement>) => {
-    const current = new Date(evt.target.value);
+  const handleClose = useCallback(() => {
+    if (isOpen) {
+      setIsOpen(false);
+    }
+  }, [isOpen, setIsOpen]);
+
+  const handleCloseOnBlur = useCallback(
+    (evt: MouseEvent) => {
+      const { relatedTarget } = evt;
+      // istanbul ignore next (this is too DOM related)
+      if (
+        component.current && relatedTarget instanceof HTMLElement
+          ? !component.current.contains(relatedTarget)
+          : true
+      ) {
+        setIsOpen(false);
+      }
+    },
+    [component, setIsOpen]
+  );
+
+  const handleInputChange = useCallback(
+    (evt: SyntheticInputEvent<HTMLInputElement>) => {
+      onChange(evt.target.value);
+    },
+    [onChange]
+  );
+
+  const handleSelect = useCallback(
+    (selected: Date) => {
+      setIsOpen(false);
+      onChange(toYYYYMMDD(selected));
+    },
+    [setIsOpen, onChange]
+  );
+
+  const handleCurrentChange = useCallback(
+    current => {
+      onCalendarChange ? onCalendarChange(current) : setCurrent(current);
+    },
+    [setCurrent, onCalendarChange]
+  );
+
+  useEffect(() => {
+    const current = new Date(value);
     // istanbul ignore else
-    if (!isNaN(current.getTime())) {
-      this.setState({ current });
+    if (!isNaN(current.getTime()) && !onCalendarChange) {
+      setCurrent(current);
     }
-    this.props.onChange(evt.target.value);
-  };
-
-  handleSelect = (selected: Date) => {
-    this.setIsOpen(false);
-    this.props.onChange(toYYYYMMDD(selected));
-  };
+  }, [value, setCurrent, onCalendarChange]);
 
-  preventDefault = (evt: Event) => evt.preventDefault();
-
-  render() {
-    const {
-      value,
-      onChange, // eslint-disable-line no-unused-vars
-      isOpen: isOpenProp, // eslint-disable-line no-unused-vars
-      onToggle, // eslint-disable-line no-unused-vars
-      min,
-      max,
-      ...props
-    } = this.props;
-
-    const { current } = this.state;
-
-    const isOpen = this.isOpen;
-
-    const dateValue = parse(value);
-    const isDateValid = !isNaN(dateValue.getTime());
+  //Sync is isOpenProp with state: isOpen
+  useEffect(() => {
```

Comment:
This won't be needed once we implement what I suggested here https://github.com/quid/refraction/pull/103/files#r317236692